java/src/com/ibm/streamsx/topology/TStream.java

```diff
@@ -89,6 +90,41 @@
      * @return Filtered stream
      */
     TStream<T> filter(Predicate<T> filter);
+    
+    /**
+     * Split a stream into {@code n} streams as specified by {@code splitter}.
+     * 
+     * <P>
+     * For each tuple on the stream, {@code splitter.applyAsInt(tuple)} is called.
+     * The return value {@code r} determines the destination stream:
+     * <pre>
+     * if r < 0 the tuple is discarded
+     * else it is sent to the stream at position (r % n) in the returned array.
+     * </pre>
+     * </P>
+     * 
+     * <P>
+     * Example of round robin splitting a stream of {@code String} into 5 channels.
```

Comment:
Missed this conversation.

I don't think split() should offer a non-function version, it is different to parallel() and I think having split(n) and parallel(n) would add confusion as to which to use.
The only way a non-function (Keyable) version would work is to use the hash code of tuple's key, it seems unrealistic that the hash code % n will match any desired routing of tuples. For any normal hash code, won't ~50% of the tuples be discarded as its value will be less than 0?